src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/CounterSampleCalculator.cs

```diff
@@ -0,0 +1,251 @@
+                //
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_100NSEC_TIMER:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_100NSEC_TIMER_INV:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_100NSEC_MULTI_TIMER:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_100NSEC_MULTI_TIMER_INV:
+                    newPdhValue.FirstValue = newSample.RawValue;
+                    newPdhValue.SecondValue = newSample.TimeStamp100nSec;
+
+                    oldPdhValue.FirstValue = oldSample.RawValue;
+                    oldPdhValue.SecondValue = oldSample.TimeStamp100nSec;
+                    if ((newCounterType & Interop.Kernel32.PerformanceCounterOptions.PERF_MULTI_COUNTER) == Interop.Kernel32.PerformanceCounterOptions.PERF_MULTI_COUNTER)
+                    {
+                        newPdhValue.MultiCount = (int)newSample.BaseValue;
+                        oldPdhValue.MultiCount = (int)oldSample.BaseValue;
+                    }
+                    break;
+                //
+                //  These counters use two data points
+                //
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_SAMPLE_FRACTION:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_RAW_FRACTION:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_LARGE_RAW_FRACTION:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_PRECISION_SYSTEM_TIMER:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_PRECISION_100NS_TIMER:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_PRECISION_OBJECT_TIMER:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_AVERAGE_TIMER:
+                case Interop.Kernel32.PerformanceCounterOptions.PERF_AVERAGE_BULK:
+                    newPdhValue.FirstValue = newSample.RawValue;
+                    newPdhValue.SecondValue = newSample.BaseValue;
+
+                    oldPdhValue.FirstValue = oldSample.RawValue;
+                    oldPdhValue.SecondValue = oldSample.BaseValue;
+                    break;
+
+                default:
+                    // an unidentified counter was returned so
+                    newPdhValue.FirstValue = 0;
+                    newPdhValue.SecondValue = 0;
+
+                    oldPdhValue.FirstValue = 0;
+                    oldPdhValue.SecondValue = 0;
+                    break;
+            }
+        }
+
+        private static void LoadPerfCounterDll()
+        {
+            if (s_perfCounterDllLoaded)
+                return;
+
+            new FileIOPermission(PermissionState.Unrestricted).Assert();
+
+            string installPath = System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory();
+            string perfcounterPath = Path.Combine(installPath, "perfcounter.dll");
+            if (Interop.Kernel32.LoadLibrary(perfcounterPath) == IntPtr.Zero)
```

Comment:
I haven't checked the code why we are loading the perfcounter.dll again. Maybe I'm missing something here.

Reply:
It appears that we load perfcounter.dll then run Interop.PerfCounter.FormatFromRawValue. Perhaps this is an instance in desktop where we need to load the library explicitly?